src/pip/_internal/network/xmlrpc.py

```diff
@@ -21,12 +24,14 @@ class PipXmlrpcTransport(xmlrpc_client.Transport):
     """
 
     def __init__(self, index_url, session, use_datetime=False):
+        # type: (str, PipSession, Optional[bool]) -> None
         xmlrpc_client.Transport.__init__(self, use_datetime)
         index_parts = urllib_parse.urlparse(index_url)
         self._scheme = index_parts.scheme
         self._session = session
 
     def request(self, host, handler, request_body, verbose=False):
+        # type: (str, str, Dict[str, str], Optional[bool]) -> None
```

Comment:
```suggestion
        # type: (str, str, Dict[str, str], bool) -> None
```

Reply:
Fixed in https://github.com/pypa/pip/pull/8271/commits/65a6152f024de93c91a152847faee925f73c050a